app/src/main/java/space/taran/arknavigator/mvp/model/repo/extra/DocumentMetaExtra.kt

```diff
@@ -0,0 +1,22 @@
+package space.taran.arknavigator.mvp.model.repo.extra
+
+import android.widget.TextView
+import space.taran.arknavigator.mvp.model.repo.index.MetaExtraTag
+import space.taran.arknavigator.mvp.model.repo.index.ResourceMetaExtra
+import space.taran.arknavigator.utils.extensions.textOrGone
+import java.nio.file.Path
+
+object DocumentMetaExtra {
+    val ACCEPTED_EXTENSIONS: Set<String> =
+        setOf("pdf", "txt", "doc", "docx", "odt", "ods", "md")
+
+    fun extract(path: Path): ResourceMetaExtra? = null
+
+    fun draw(extra: ResourceMetaExtra, pagesTV: TextView, verbose: Boolean) {
+        val pages = extra.data[MetaExtraTag.PAGES]
+        if (pages != null) {
+            val label = if (verbose) "$pages pages" else "$pages"
+            pagesTV.textOrGone(label)
+        }
```

Comment:
Maybe the text generation can be left here, but the work with the view should be in the `ui` package
Same for `ImageMetaExtra` and `VideoMetaExtra`

Reply:
Yes, you are right. Feel free to change this.